source/module_io/get_wf_lcao.h

```diff
@@ -120,5 +131,8 @@ class IState_Envelope
     template <typename T>
     void wfc_2d_to_grid(const T* wfc_2d, const Parallel_Orbitals& pv, T** wfc_grid, const std::vector<int>& trace_lo);
 #endif
+
+    std::vector<int> bands_picked_;
+    const elecstate::ElecState* pes_ = nullptr;
```

Comment:
I have some experience to share:
only you really want the instance of one class has some "states", you let it own some data member, otherwise not, and better not use a class, especially when the meaning of instance is not clear, for example, could you tell me what is the meaning of instance of class "IState_Envelope"? I am sure you will find it is nearly nonsence to "instantiate IState_Envelope" because the instance of it is totally meaningless...
So a practical question is, why you add these two data members? are they necessary, and should be included in the state of instance of this class throughout its life span?